src/_nebari/template/stages/07-kubernetes-services/modules/kubernetes/services/jupyterhub/files/jupyterhub/03-profiles.py

```diff
@@ -311,6 +311,61 @@ def configure_user(username, groups, uid=1000, gid=100):
     }
 
 
+def profile_argo_token(groups):
+    # TODO: create a more robust check user's Argo-Workflow role
+
+    domain = z2jh.get_config("custom.external-url")
+
+    ADMIN = "admin"
+    DEVELOPER = "developer"
+    ANALYST = "analyst"
+
+    base = "argo-"
+
+    if ANALYST in groups:
+        argo_sa = base + "view"
+    if DEVELOPER in groups:
+        argo_sa = base + "dev"
+    if ADMIN in groups:
+        argo_sa = base + "admin"
+    else:
+        return {}
+
+    return {
+        "ARGO_BASE_HREF": "/argo",
+        "ARGO_SERVER": f"{domain}:443",
+        "ARGO_TOKEN": {
+            "valueFrom": {
+                "secretKeyRef": {
+                    "name": f"{argo_sa}.service-account-token",
```

Comment:
Did the modifications you made create this token or did it already exist?

Reply:
This token already exists, it's the default `argo-admin`, `argo-dev` and `argo-view` secrets. I'm not entirely sure if they were being used prior to this... The ARGO_TOKEN used when someone creates a workflow from the UI (and the one that can be copied from there) are not the same as any of those above. 

I have also included this [the enhancements issue](https://github.com/nebari-dev/nebari/issues/1838), namely, the ability to create short-lived tokens on a per user basis.